app/views/posts/show.html.erb

```diff
@@ -10,4 +10,8 @@
     </div>
     <%= render partial: 'shared/comments', locals: { comments: @comments } %>
   </div>
+  <%= render 'shared/comment_form' %>
+  <div class="flex justify-end -mt-10 ">
+  <%= render 'shared/like' %>
```

Comment:
- [ ] Good job for adding a comment form and creating partials ✨However, please make sure that there is a form in the `views/comments/new.html.erb` file which accepts input for creating comments. Currently, a form for adding comments is rendering in `views/posts/show.html.erb` However, it is violating the principles of separation of concerns and modularity in web development 📌⛏ Each component of your application (models, views, controllers) should have a clear and distinct responsibility.

   - _The `posts/show.html.erb` view is meant for displaying information about a single post, not for creating new posts/comments._
   - _The `comments/new.html.erb` view, on the other hand, should be responsible for rendering the form to create a new comment._

  Mixing these responsibilities can make the codebase less maintainable and harder to understand✨ So, please make sure to keep the comment creation form in its own dedicated view (`views/comments/new.html.erb`)

Reply:
Hi @KanzaTahreem, I understand what you mean, I understand perfectly the convention which is why I have reached this point, but I do not agree that it violates the principle of separation of responsibilities and modularization.

First, because the form that creates the comment is perfectly modularized and its logic is completely separate in another component.
Second, I think that I have not violated the MVC pattern, nor have I mixed responsibilities since the only thing I have done was to put a view in a partial to avoid unnecessary rendering.
Third, because I have not seen a single blog or social network that to create a comment on a post render a new page, the rule to preserve a good user experience is to create the comments on the same page, avoiding unnecessary rendering, as you can see even here on **GitHub**.
 So what is more important, to follow the industry conventions and look for a good user experience or to follow strictly the rails conventions?

Reply:
### Project requirements
-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
    -  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.